### PR TITLE
Restaurant owners can create a restaurant

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT Story: https://www.pivotaltracker.com/story/show/131526907

Changes proposed in this pull request:
- Added restaurant creation page, routing, controller
- Restrict restaurant creation to logged in restaurant owners & error handle for exception cases
- Create simple show page to display restaurant information (but this needs to be expanded in another feature)
- Require address and name for restaurant

Todo:
- Complete sad path testing for restaurant creation - no name, no associated user, etc. (Error messages should be created and should appear but we haven't tested for them for these scenarios).

What I have learned working on this feature:
- How to make changes to DB tables
- merging parameters into table entry
- CanCan's load_and_authorize_resource

Screenshots:
screen shot 2016-10-03 at 3 02 24 pm

Ready for review!

### CODE REVIEW

File: app/controllers/restaurants_controller.rb [hunk 1]
```diff
@@ -1,9 +1,29 @@
 class RestaurantsController < ApplicationController
+  load_and_authorize_resource # potentially refactor this into ApplicationController
```
Comment (tochman) [inline app/controllers/restaurants_controller.rb:2]: Try this refactoring and see if all test pass.

File: app/controllers/restaurants_controller.rb [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
 
+  def create
+    @restaurant = Restaurant.new(restaurant_params.merge({user: current_user}))
+    if @restaurant.save
+      render :show
+    else
+      flash[:alert] = @restaurant.errors.full_messages.first
+      render :new
+    end
+  end
```
Comment (tochman) [inline app/controllers/restaurants_controller.rb:20]: The creation of a restaurant should be limited if the user already has a restaurant

File: features/restaurant_page.feature [hunk 3]
```diff
@@ -2,14 +2,36 @@ Feature: As a restaurant Owner
...
+    | Zipcode     | 41235                |
+    | Town        | Göteborg             |
+  And I click the "Create" button
+  And I should see "Street can't be blank"
+
+Scenario: I attempt to access new restaurant page without being logged in
+  Given I am not logged in
+  When I am on the "create restaurant" page
+  Then I should be on the "index" page
+  Then I should see "You are not authorized to access this page"
```
Comment (tochman) [inline features/restaurant_page.feature]: `And I should...`

**Review by tochman — CHANGES_REQUESTED** [review]
Looks very good. Small comments.....

File: app/controllers/restaurants_controller.rb (hunk 1 above)
Comment (AmberWilkie) [reply app/controllers/restaurants_controller.rb:2]: It didn't.

File: app/controllers/restaurants_controller.rb (hunk 2 above)
Comment (AmberWilkie) [reply app/controllers/restaurants_controller.rb:20]: Done.

File: features/restaurant_page.feature (hunk 3 above)
Comment (AmberWilkie) [reply features/restaurant_page.feature]: Done.

**Review by tochman — APPROVED** [review]

**Pull request merged**